### PR TITLE
Apns2 fixes for handling CONNRESET and priority

### PR DESCRIPTION
**Pull request by rajesh-blueshift**
We are seeing a bunch of connection reset errors with Rpush.
If there is such an error, the entire batch gets discarded without marking it as failed or succeeded.
Handled the connection reset error and attempted to retry the batch on occurrence of such an error.

Also backported a few fixes that I felt were relevant:
https://github.com/rpush/rpush/pull/567/files
https://github.com/rpush/rpush/pull/579/files

### CODE REVIEW

File: lib/rpush/daemon/apns2/delivery.rb [hunk 1]
```diff
@@ -111,6 +111,18 @@ def prepare_headers(notification)
           if headers['apns-topic'].blank? && @app.bundle_id.present?
             headers['apns-topic'] = @app.bundle_id
           end
+
+          headers['apns-expiration'] = '0'
```
Comment (chongbomb) [inline lib/rpush/daemon/apns2/delivery.rb:115]: The headers in `notification_data` should take precedence, take a look at the implementation in https://github.com/rpush/rpush/pull/579/files

**Review by chongbomb — CHANGES_REQUESTED** [review]
The base branch we use is actually redis_cluster_4.2.0_fixes. Can you update to reflect that?

**Comment by rajesh-blueshift** [comment]
> The base branch we use is actually redis_cluster_4.2.0_fixes. Can you update to reflect that?

Sure

**Comment by rajesh-blueshift** [comment]
@chongbomb , I made the CONNRESET retries bounded now. If it exceeds, then the push will be marked failed.

File: lib/rpush/daemon/apns2/delivery.rb (hunk 1 above)
Comment (rajesh-blueshift) [reply lib/rpush/daemon/apns2/delivery.rb:115]: Done

**Review by chongbomb — APPROVED** [review]

**Pull request merged**